src/main/java/fopbot/Robot.java

```diff
@@ -303,6 +303,16 @@ public int getNumberOfCoins() {
         return numberOfCoins;
     }
 
+    /**
+     * Sets the current number of coins this robot has.
+     *
+     * @param coins The current number of coins this robot has
+     */
+    public void setNumberOfCoins(int coins) {
+        world.trace(this, RobotAction.SET_NUMBER_OF_COINS);
+        this.numberOfCoins = coins;
```

Comment:
In constructor, `world.checkNumberOfCoins(numberOfCoins)` is called to avoid negative numbers of coins. In my opinion, this should also be done `setNumberOfCoins`.